packages/core/index.d.ts

```diff
@@ -48,27 +48,21 @@ export interface MiddyfiedHandler<TEvent = any, TResult = any, TErr = Error, TCo
     after: Array<MiddlewareFn<TEvent, TResult, TErr, TContext>>
     onError: Array<MiddlewareFn<TEvent, TResult, TErr, TContext>>
   }
-  (event: TEvent, context: TContext): Promise<TResult>
 }
 
-declare type AttachMiddlewareFn<TEvent = any, TResult = any, TErr = Error, TContext extends LambdaContext = any> = (middleware: MiddlewareFn) => MiddyfiedHandler<TEvent, TResult, TErr, TContext>
+declare type AttachMiddlewareFn<TEvent = any, TResult = any, TErr = Error, TContext extends LambdaContext = LambdaContext> = (middleware: MiddlewareFn) => MiddyfiedHandler<TEvent, TResult, TErr, TContext>
 
-declare type AttachMiddlewareObj<TEvent = any, TResult = any, TErr = Error, TContext extends LambdaContext = any> = (middleware: MiddlewareObj) => MiddyfiedHandler<TEvent, TResult, TErr, TContext>
+declare type AttachMiddlewareObj<TEvent = any, TResult = any, TErr = Error, TContext extends LambdaContext = LambdaContext> = (middleware: MiddlewareObj) => MiddyfiedHandler<TEvent, TResult, TErr, TContext>
 
-declare type UseFn<TEvent = any, TResult = any, TErr = Error, TContext extends LambdaContext = any> =
+declare type UseFn<TEvent = any, TResult = any, TErr = Error, TContext extends LambdaContext = LambdaContext> =
   (middlewares: MiddlewareObj<TEvent, TResult, TErr, TContext> | Array<MiddlewareObj<TEvent, TResult, TErr, TContext>>) => MiddyfiedHandler<TEvent, TResult, TErr, TContext>
 
-declare type MiddlewareHandler<THandler extends LambdaHandler<any, any>, TContext> =
-  THandler extends LambdaHandler<infer TEvent, infer TResult> // always true
```

Comment:
It's nice to have this type joined to the one exported from the AWS typings in some way. Maybe we can test they intersect in the typings test? Not a big deal either way given these types are very very unlikely to change